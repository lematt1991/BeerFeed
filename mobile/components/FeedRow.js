import React from 'react';
import {
	Text,
	StyleSheet,
	View,
	Image
} from 'react-native'

const styles = StyleSheet.create({
	container : {
		flex : 1,
		flexDirection : 'row',
		alignItems : 'center',
		justifyContent: 'flex-start',
		marginBottom : 10
	},
	imageContainer : {
		marginRight : 10
	},
	image : {
		width : 50,
		height : 50
	}
})

export default class FeedRow extends React.Component{
	render(){
		var dateFormat = {
			year : '2-digit',
			month : '2-digit',
			day : '2-digit',
			hour : '2-digit',
			minute : '2-digit'
		}
		var date = new Date(this.props.created)
		date.setMinutes(date.getMinutes() - date.getTimezoneOffset()) //convert from UTC to local timezone
		date.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
		var beerLink = `https://untappd.com/b/${this.props.beer_slug}/${this.props.bid}`
		var dateFormat = {
			year : '2-digit',
			month : '2-digit',
			day : '2-digit',
			hour : '2-digit',
			minute : '2-digit'
		}
		var date = new Date(this.props.created)
		date.setMinutes(date.getMinutes() - date.getTimezoneOffset()) //convert from UTC to local timezone
		date.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
		return(
			<View style={styles.container}>
				<View style={styles.imageContainer}>	
					<Image 
						source={{uri : this.props.pic}}
						style={styles.image}
						defaultSource={require('../assets/badge-beer-default.png')}
					/>
				</View>
				<View>
					<Text>
						Beer: {this.props.name}
					</Text>
					<Text>
						Brewery: {this.props.brewery}
					</Text>
					<Text>
						Found at: {this.props.venue}
					</Text>
					<Text>
						Rating: {this.props.rating}
					</Text>
					<Text>
						Number of checkins: {this.props.checkin_count}
					</Text>
					<Text>
						Last checked in: {date.toLocaleString([], dateFormat)}
					</Text>
				</View>
			</View>
		)
	}
}

