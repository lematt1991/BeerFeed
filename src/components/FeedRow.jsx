import React, {Component} from 'react';
import * as _ from 'lodash'
import { SafeAnchor } from 'react-bootstrap';
import { push } from 'react-router-redux'

export default class FeedRow extends Component{
	constructor(props){
		super(props);
		this.state = {
			picSrc : props.beer.pic,
			error : false
		}
	}

	toMap = (venueID) => {
		this.props.dispatch(push({
			pathname : `/map/${venueID}`,
		}))
	}

	handleError = () => {
		if(!this.state.error){
			this.setState(_.extend({}, this.state, {
				picSrc : 'https://untappd.akamaized.net/site/assets/images/temp/badge-beer-default.png',
				error : true
			}))
		}
	}

	render(){
		var dateFormat = {
			year : '2-digit',
			month : '2-digit',
			day : '2-digit',
			hour : '2-digit',
			minute : '2-digit'
		}
		var date = new Date(this.props.created)
		date.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit'});
		var beerLink = `https://untappd.com/b/${this.props.beer.slug}/${this.props.bid}`
		return(	
			<div className="media">
				<div className="media-left media-middle">
					<img
						alt='' 
						onError={this.handleError}
						style={{marginTop : '12px'}} 
						src={this.state.picSrc} 
						width="100" 
						height="100" 
						className="media-photo"
					/>
				</div>
				<div className="media-body">
					<span className="media-meta pull-right">{date.toLocaleString([], dateFormat)}</span>
					<h4 className="title">
						Brewery: {this.props.brewery}
					</h4>
					<h4>
						Beer: <a target="_blank" href={beerLink}>{this.props.beer.name}</a>
					</h4>
					<h4>
						Score: {this.props.beer.rating}
					</h4>
					<h4>	
						Found at: <SafeAnchor id='goto-map'
									onClick={() => this.toMap(this.props.venue_id)}>{this.props.venue}</SafeAnchor>
					</h4>
					<h4>
						Number of checkins: {this.props.checkin_count}
					</h4>
					<h4>
						Style: {this.props.beer.style}
					</h4>
				</div>
			</div>
		);
	}
}