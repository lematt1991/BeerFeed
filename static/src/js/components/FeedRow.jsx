import React, {Component} from 'react';
import * as _ from 'lodash'
import {SafeAnchor} from 'react-bootstrap';


export default class FeedRow extends Component{
	constructor(props){
		super(props);
	}

	render(){
		var date = new Date(this.props.created)
		var beerLink = `https://untappd.com/b/${this.props.beer_slug}/${this.props.bid}`
		return(
			<tr data-status="pagado" key={this.props.checkin_id}>
				<td>
					<div class="media">
						<img style={{marginTop : '12px'}} src={this.props.pic} width="100" height="100" class="pull-left media-photo beer-image"/>
						<div class="media-body">
							<span class="media-meta pull-right">{date.toDateString()}</span>
							<h4 class="title">
								Brewery: {this.props.brewery}
							</h4>
							<h4>
								Beer: <a target="_blank" href={beerLink}>{this.props.name}</a>
							</h4>
							<h4>
								Score: {this.props.rating}
							</h4>
							<h4>	
								Found at: <SafeAnchor 
											onClick={() => this.props.history.pushState({pos : {lat:this.props.lat, lng:this.props.lon}, venue : this.props.venue}, 'map/' + this.props.venue_id)}>{this.props.venue}</SafeAnchor>
							</h4>
						</div>
					</div>
				</td>
			</tr>
		);
	}
}