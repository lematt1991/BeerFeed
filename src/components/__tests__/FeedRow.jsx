import React from 'react'
import Enzyme, {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import sinon from 'sinon'
import Adapter from 'enzyme-adapter-react-16'

import FeedRow from '../FeedRow'
import {Checkins1} from '../../test_data/Checkins'

Enzyme.configure({ adapter: new Adapter() });

describe('<FeedRow/>', () => {
	var toLocaleString;

	beforeAll(() => {
		toLocaleString = sinon.stub(Date.prototype, 'toLocaleString').callsFake(() => 'fake time')
	})

	afterAll(() => {
		toLocaleString.restore()
	})

	it('Mounts', () => {
		const context = {
			router : {
				push : jest.fn()
			}
		}
		var wrapper = shallow(<FeedRow {...Checkins1.checkins[0]}/>, {context})
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('Goes to the map when a venue is clicked', () => {
		const dispatch = jest.fn()

		var checkin = Checkins1.checkins[0]
		var wrapper = shallow(<FeedRow {...checkin} dispatch={dispatch}/>)
		wrapper.find('#goto-map').simulate('click')

		var arg = dispatch.mock.calls[0][0]
		expect(arg).toMatchObject({
		  "payload": {
		    "args": [{"pathname": "/map/5030814"}],
		    "method": "push"
		  },
		  "type": "@@router/CALL_HISTORY_METHOD"
		})
	})
})