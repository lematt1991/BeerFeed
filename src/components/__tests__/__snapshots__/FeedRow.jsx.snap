// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedRow/> Mounts 1`] = `
<div
  className="media"
>
  <div
    className="media-left media-middle"
  >
    <img
      alt=""
      className="media-photo"
      height="100"
      onError={[Function]}
      src="https://untappd.akamaized.net/site/beer_logos/beer-1501561_e5172_sm.jpeg"
      style={
        Object {
          "marginTop": "12px",
        }
      }
      width="100"
    />
  </div>
  <div
    className="media-body"
  >
    <span
      className="media-meta pull-right"
    >
      fake time
    </span>
    <h4
      className="title"
    >
      Brewery: 
      Half Acre Beer Company
    </h4>
    <h4>
      Beer: 
      <a
        href="https://untappd.com/b/half-acre-beer-company-deep-space/1501561"
        target="_blank"
      >
        Deep Space
      </a>
    </h4>
    <h4>
      Score: 
      4.17661
    </h4>
    <h4>
      Found at: 
      <SafeAnchor
        componentClass="a"
        id="goto-map"
        onClick={[Function]}
      >
        Maple St
      </SafeAnchor>
    </h4>
    <h4>
      Number of checkins: 
      1
    </h4>
    <h4>
      Style: 
      IPA - Imperial / Double
    </h4>
  </div>
</div>
`;
